plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -952,9 +952,36 @@ public void createSchema(ConnectorSession session, String schemaName, Map<String
     }
 
     @Override
-    public void dropSchema(ConnectorSession session, String schemaName)
+    public void dropSchema(ConnectorSession session, String schemaName, boolean cascade)
     {
-        metastore.dropDatabase(session, schemaName);
+        if (cascade) {
```

Comment:
In case of hive configured with a thrift metastore it has a dedicated API for cascade. Any reason for not using it ?

Reply:
The option doesn't work when there're non-Hive tables (e.g. Iceberg). It deletes tables from metastore without removing data files.

Reply:
Can we capture that test case so that in the future we might not optimize it ? And also to verify the state of the schema and tables of the operation like if it failed in the middle ?